server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java

```diff
@@ -111,19 +116,29 @@ private Mailbox fromMessageManager(MessageManager messageManager, Optional<List<
             .unreadMessages(mailboxCounters.getUnseen())
             .totalMessages(mailboxCounters.getCount())
             .sortOrder(SortOrder.getSortOrder(role))
-            .sharedWith(Rights.fromACL(metaData.getACL()))
-            .namespace(getNamespace(mailboxPath, mailboxSession))
+            .sharedWith(rights)
+            .mayAddItems(rights.mayAddItems(username).orElse(isOwner))
+            .mayCreateChild(rights.mayCreateChild(username).orElse(isOwner))
+            .mayDelete(rights.mayDelete(username).orElse(isOwner))
+            .mayReadItems(rights.mayReadItems(username).orElse(isOwner))
+            .mayRemoveItems(rights.mayRemoveItems(username).orElse(isOwner))
+            .mayRename(rights.mayRename(username).orElse(isOwner))
+            .namespace(getNamespace(mailboxPath, isOwner))
             .build();
     }
 
-    private MailboxNamespace getNamespace(MailboxPath mailboxPath, MailboxSession mailboxSession) {
-        String mailboxPathUser = mailboxPath.getUser();
-        if (mailboxSession.getUser().isSameUser(mailboxPathUser)) {
+    private MailboxNamespace getNamespace(MailboxPath mailboxPath, boolean isOwner) {
```

Comment:
this method is just a less generic `if` (:
I mean, it only choose from two simple values based on a boolean you pass.

Reply:
I don't agree. It also computes the owner name. It also extracts Namespace generation from a WAAAAY too long method, thus it make sens.

BTW this extraction makes sens as we might imagine making this object more complex in the future:
 - personal
 - delegated (user)
 - shared (account)
 - external(provider)
 - etc...

Thus **this extraction makes sens**.